### PR TITLE
Implemented Filecoin.MsigGetAvailableBalance API

### PR DESCRIPTION
**Pull request by sudo-shashank**
## Summary of changes

Changes introduced in this pull request:

- Added support for Filecoin.MsigGetAvailableBalance API
```
| RPC Method                       | Forest | Lotus |
|----------------------------------|--------|-------|
| Filecoin.MsigGetAvailableBalance | Valid  | Valid |
```

## Reference issue to close (if applicable)

As part of https://github.com/ChainSafe/forest/issues/3639

## Change checklist

- [X] I have performed a self-review of my own code,
- [X] I have made corresponding changes to the documentation,
- [X] I have added tests that prove my fix is effective or that my feature works (if possible),
- [X] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/rpc/state_api.rs [hunk 1]
```diff
@@ -670,6 +670,24 @@ pub(in crate::rpc) async fn state_circulating_supply<DB: Blockstore + Send + Syn
...
+    data: Data<RPCState<DB>>,
+    Params(LotusJson((addr, tsk))): Params<LotusJson<(Address, TipsetKeys)>>,
+) -> Result<LotusJson<TokenAmount>, JsonRpcError> {
+    let ts = data.chain_store.load_required_tipset(&tsk)?;
+    let height = ts.epoch();
+    let store = data.state_manager.blockstore();
+    let actor = data
+        .state_manager
+        .get_actor(&addr, *ts.parent_state())?
+        .ok_or("MultiSig actor not found".to_string())?;
```
Comment (LesnyRumcajs) [inline src/rpc/state_api.rs:683]:
```suggestion
        .ok_or("MultiSig actor not found")?;
```

File: src/rpc/state_api.rs [hunk 2]
```diff
@@ -670,6 +670,24 @@ pub(in crate::rpc) async fn state_circulating_supply<DB: Blockstore + Send + Syn
...
+    let actor_balance = TokenAmount::from(actor.balance.clone());
```
Comment (LesnyRumcajs) [inline src/rpc/state_api.rs:684]:
```suggestion
    let actor_balance = TokenAmount::from(&actor.balance);
```

File: src/rpc/state_api.rs [hunk 3]
```diff
@@ -670,6 +670,24 @@ pub(in crate::rpc) async fn state_circulating_supply<DB: Blockstore + Send + Syn
...
+    let ms = multisig::State::load(&store, actor.code, actor.state)?;
+    let locked_balance = ms.locked_balance(height)?.into();
+    let avail_balance = actor_balance.clone() - &locked_balance;
```
Comment (LesnyRumcajs) [inline src/rpc/state_api.rs:687]:
```suggestion
    let avail_balance = &actor_balance - locked_balance;
```

**Review by LesnyRumcajs — COMMENTED** [review]
Let's avoid cloning when possible (especially when it makes the code more succinct)

File: src/rpc/state_api.rs (hunk 1 above)
Comment (sudo-shashank) [reply src/rpc/state_api.rs]: corrected

File: src/rpc/state_api.rs (hunk 2 above)
Comment (sudo-shashank) [reply src/rpc/state_api.rs]: corrected

File: src/rpc/state_api.rs (hunk 3 above)
Comment (sudo-shashank) [reply src/rpc/state_api.rs]: corrected

**Review by LesnyRumcajs — APPROVED** [review]

**Review by jdjaustin — APPROVED** [review]

**Pull request merged**